### PR TITLE
Add sonarqube task

### PR DESCRIPTION
**Pull request by AsfaMumtaz**

### CODE REVIEW

**Comment by stakater-user** [comment]
@AsfaMumtaz Validation Successful

File: clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          fi
+        
+          replaceValues $(workspaces.source.path)/sonar-project.properties sonar.login $sonar_login
+          replaceValues $(workspaces.source.path)/sonar-project.properties sonar.password $sonar_password
+        
+        else
+          echo "No sonar-project.properties found"
+          exit 128
+        fi
+
```
Comment (rasheedamir) [inline clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml:77]: @AsfaMumtaz remove empty line

File: clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        - /bin/bash
+      args:
+        - -c
+        - |
+          if [ -f "pom.xml" ]
+          then
+            mvn -f pom.xml clean package
+          fi
+          sonar-scanner
+
```
Comment (rasheedamir) [inline clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml:92]: @AsfaMumtaz too many empty lines

File: clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      description: Project's unique key
+      default: ""
+    - name: sonar_login
+      description: Sonar username
+      default: ""
+    - name: sonar_password
+      description: Sonar password
+      default: ""
+  steps:
+    - name: sonar-properties-create
```
Comment (rasheedamir) [inline clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml:31]: @AsfaMumtaz name with -? doesn't seem to be following our conventions

Comment (AsfaMumtaz) [reply clustertask/stakater-sonarqube-scanner/templates/clustertask.yaml:31]: I think the - convention is fine for step name. I've changed all params to follow the underscore + capitalized convention

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**